runtime/vm/lookupmethod.c

```diff
@@ -1031,10 +1031,81 @@ illegalAccessMessage(J9VMThread *currentThread, IDATA badMemberModifier, J9Class
 	char *destModuleMsg = NULL;
 	char packageNameBuf[J9VM_PACKAGE_NAME_BUFFER_LENGTH];
 	char *packageNameMsg = NULL;
+#if defined(J9VM_OPT_VALHALLA_NESTMATES)
+	J9Class *unverifiedNestMemberClass = NULL;
+#endif /* J9VM_OPT_VALHALLA_NESTMATES */
 	
 	PORT_ACCESS_FROM_VMC(currentThread);
 	Trc_VM_illegalAccessMessage_Entry(currentThread, J9UTF8_LENGTH(senderClassNameUTF), J9UTF8_DATA(senderClassNameUTF),
 			J9UTF8_LENGTH(targetClassNameUTF), J9UTF8_DATA(targetClassNameUTF), badMemberModifier);
+
+#if defined(J9VM_OPT_VALHALLA_NESTMATES)
+	/* Values of error codes for nestmates visibility checks on source class is
+	 * odd and on destination/target class is even
+	 */
+	if ((J9_VISIBILITY_NEST_HOST_LOADING_FAILURE_ERROR_SOURCE_CLASS <= errorType)
+			&& (J9_VISIBILITY_NEST_MEMBER_NOT_CLAIMED_ERROR_DEST_CLASS >= errorType)) {
+		if (0 == errorType % 2) {
+			unverifiedNestMemberClass = targetClass;
+		} else if (1 == ((UDATA)targetClass->nestHost & 1)) {
```

Comment:
do you still need to check this? if errorType is an odd value is the error always sender?